test/indexer/test_hca_indexer.py

```diff
@@ -1270,23 +1270,44 @@ def test_cell_counts(self):
         self._index_canned_bundle(bundle_fqid)
         hits = self._get_all_hits()
         expected_cell_counts = {
-            'estimated_cell_count': 10000,  # 1 project
-            'total_estimated_cells': 20000 + 20000,  # 2 cell_suspensions
-            'matrix_cell_count': 2100 + 15000,  # 2 files
+            ('projects', 'estimated_cell_count'): {
+                '90bf705c-d891-5ce2-aa54-094488b445c6': 10000
+            },
+            ('cell_suspensions', 'total_estimated_cells'): {
+                'd9eaaffe-4c93-5503-984f-762e8dfddce4': 20000,
+                'af3549f4-8632-58c2-9821-c6249c622b0a': 20000,
+            },
+            ('files', 'matrix_cell_count'): {
+                'cf5d9300-3b81-52eb-a02f-25fb1364419e': 2100,
+                'f0c8ac7c-1c4a-5c9c-9286-c8b16208f9ae': 15000
+            }
         }
-        actual_cell_counts = {}
+        actual_hits = defaultdict(int)
         for hit in hits:
             contents = hit['_source']['contents']
-            if ('projects', True) == self._parse_index_name(hit):
-                project = one(contents['projects'])
-                cell_suspension = one(contents['cell_suspensions'])
-                file = one(contents['files'])
-                actual_cell_counts = {
-                    'estimated_cell_count': project['estimated_cell_count'],
-                    'total_estimated_cells': cell_suspension['total_estimated_cells'],
-                    'matrix_cell_count': file['matrix_cell_count'],
-                }
-        self.assertEqual(expected_cell_counts, actual_cell_counts)
+            entity_type, aggregate = self._parse_index_name(hit)
+            actual_hits[(entity_type, aggregate)] += 1
+            if aggregate:
+                if entity_type == 'projects':
+                    for key, expected_counts in expected_cell_counts.items():
+                        entity_type, field_name = key
+                        entity = one(contents[entity_type])
+                        expected_sum = sum(expected_counts.values())
+                        self.assertEqual(entity[field_name], expected_sum)
```

Comment:
It's not immediately apparent that this assertion is guaranteed to be hit. So it looks like yet another instance of aconditional/loop body assertion.